src/main/java/organice/logic/commands/MatchCommand.java

```diff
@@ -0,0 +1,90 @@
+package organice.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import organice.model.Model;
+import organice.model.person.BloodType;
+import organice.model.person.Donor;
+import organice.model.person.Nric;
+import organice.model.person.Patient;
+import organice.model.person.Person;
+import organice.model.person.TissueType;
+import organice.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * Lists all persons in the address book to the user.
+ */
+public class MatchCommand extends Command {
+
+    public static final String COMMAND_WORD = "match";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Matches patients and donors who passes matching tests"
+            + "Parameters: all/ (to match all patients and donors)"
+            + "Parameters: ic/NRIC (to match a patient of specified NRIC)";
+
+    public static final String MESSAGE_SUCCESS = "Matched all patients and donors";
+    public static final String MESSAGE_PERSON_NOT_FOUND = "The patient with Nric %1$s cannot be found in ORGANice!";
+
+    public static final Double SUCCESSFUL_PERCENTAGE = 80.0;
+
+
+    private String input;
+    private Patient patient;
+
+    public MatchCommand(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Returns true if the donor and patient is a match.
+     * A match happens when the donor and patient's blood type and tissue types are compatible.
+     * @param donor {@code Donor} who will tested for suitability of donation
+     * @param patient {@code Patient} in need of organ
+     */
+    public boolean match(Person donor, Patient patient) {
+        if (!(donor instanceof Donor)) {
+            return false;
+        }
+
+        BloodType donorBloodType = ((Donor) donor).getBloodType();
+        BloodType patientBloodType = patient.getBloodType();
+
+        TissueType donorTissueType = ((Donor) donor).getTissueType();
+        TissueType patientTissueType = patient.getTissueType();
+
+        if (patientBloodType.isBloodTypeMatch(donorBloodType)
+                && donorTissueType.getPercentageMatch(patientTissueType) >= SUCCESSFUL_PERCENTAGE) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+
+        Nric patientNric = new Nric(input);
+        try {
+            if (model.hasPatient(patientNric)) {
+                patient = model.getPatient(patientNric);
+                Predicate<Person> matchesWithNric = donor -> match(donor, patient);
+                model.updateFilteredPersonList(matchesWithNric);
```

Comment:
Need to consider showing the patient as well more on the ui side like on top show the patient and below all the potential donors

Reply:
one more thing consider no match found scenario as well, do you want to throw error or what?

Reply:
agree with william in the no match scenario. perhaps can throw a string that says there is currently no match

Reply:
Amended